packages/expect/src/utils.js

```diff
@@ -181,22 +186,30 @@ export const iterableEquality = (a: any, b: any) => {
         }
       }
       if (allFound) {
+        count = 0;
         return true;
       }
     }
   }
 
   const bIterator = b[IteratorSymbol]();
-
   for (const aValue of a) {
```

Comment:
Can you put `count` and `iterableControler` (also typo, but call it `maxIterations` or something) here, instead of at the top level?
If that messes up, at least put `count` at the top of `iterableEquality` so that we avoid the value leaking across assertions 🙂

Reply:
Sure, I will do it right away! Thanks for the feedback and the help. Any ideas why it didn't pass through all the checks?

Reply:
Reset it for every time the equality function is called